### PR TITLE
cilium: IPAMENI may add interfaces at runtime add encryption support

### PR DESCRIPTION
**Pull request by jrfastab**
EKS may add interfaces at runtime, but when this happens and encryption is enabled we fail to attach our decryption program to the new interfaces. If we receive encrypted traffic on this interface we would then fail to decrypt it and it would be dropped. Fix by adding encryption logic to update hook.

Depends on:
https://github.com/cilium/cilium/pull/15669

### CODE REVIEW

**Comment by jrfastab** [comment]
Marking as a draft until dependent PR is merged and we can rebase. Only patch 3 here is relevant.

File: pkg/ipam/crd_eni.go [hunk 1]
```diff
@@ -200,5 +205,20 @@ func configureENINetlinkDevice(link netlink.Link, cfg eniDeviceConfig) error {
 		return fmt.Errorf("failed to delete default route %q on link %q: %w", cfg.ip, link.Attrs().Name, err)
 	}
 
+	if option.Config.EnableIPSec {
+		interfaces := []string{cfg.name}
+		ctx, cancel := context.WithTimeout(context.TODO(), defaults.ExecTimeout)
```
Comment (gandro) [inline pkg/ipam/crd_eni.go]: nit:
```suggestion
		ctx, cancel := context.WithTimeout(context.Background(), defaults.ExecTimeout)
```

`context.TODO()` and `context.Background()` are functionally absolutely equivalent, but given we simply want to create a timed context here, using Background seems reasonable to me.

**Review by gandro — APPROVED** [review]
Looks good to me, at least from the ENI/CRD side (I'm less familiar with the IPSec requirements)

**Comment by nbusseneau** [comment]
Is this the fix for the recurrent hangs on `ConformanceEKS` with encryption enabled, such as here? https://github.com/cilium/cilium/actions/runs/752152075

**Comment by jrfastab** [comment]
> Is this the fix for the recurrent hangs on `ConformanceEKS` with encryption enabled, such as here? https://github.com/cilium/cilium/actions/runs/752152075

This is unrelated. I'm not sure that has anything to do with encryption or is just a `cilium uninstall/reinstall` issue.

**Comment by joestringer** [comment]
@jrfastab is this PR necessary now that we have merged https://github.com/cilium/cilium/pull/15867 ?

**Comment by pchaigno** [comment]
@jrfastab Do we have an issue to track this?

**Pull request closed without merging**